compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/dfa/cfg/ControlFlowGraphBuilder.kt

```diff
@@ -283,7 +283,7 @@ class ControlFlowGraphBuilder {
         if (invocationKind.hasTowardEdge) {
             addEdge(enterNode, exitNode)
         }
-        if (invocationKind.hasBackEdge) {
+        if (invocationKind == null || invocationKind.hasBackEdge) {
```

Comment:
According to `isInPlace` and `hasBackEdge`, null invocation kind could be deemed as an UNKNOWN event occurrence. Then, should we do the same thing for toward edge above?

Reply:
Oh, so you meant, adding a toward edge (to the entrance of a lambda) means this lambda is invoked in-place. Makes sense.

Then, is it better to use `!invocationKind.isInPlace` here to be clearer?

Reply:
I think it's more confusing to use `isInPlace`. Whether there should be a back edge has nothing to do with whether the call is inplace. The code would work if I check `isInPlace` only because currently we don't have more granular contract for non-inplace cases.

But I guess it makes sense to move the check to `hasBackEdge`.

Reply:
Yeah, having that in `hasBackEdge` looks better!